core/src/main/java/me/aurium/branch/annotate/marker/permission/Permission.java

```diff
@@ -0,0 +1,34 @@
+/*
+ *
+ *  Branch
+ *  Copyright © 2021 Aurium
+ *
+ *  Branch is free software: you can redistribute it and/or modify
+ *  It under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  Branch is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with Branch. If not, see <https://www.gnu.org/licenses/>
+ *  and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package me.aurium.branch.annotate.marker.permission;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD,ElementType.CONSTRUCTOR})
+public @interface Permission {
+
+    Class<? extends StringPermChecker> getRelativeChecker();
```

Comment:
```suggestion
    Class<? extends StringPermChecker> relativeChecker();
```
Do you really want your annotation to look like `@Permission(permission = "perm", getRelativeChecker = CheckerImpl.class)`?

Reply:
What do you suggest as an alternative? @Permission and an optional @PermissionChecker? (Since the default checker is obviously determined by the platform's parser e.g. minecraft = MinecraftChecker.class